guides/k8-akka-cluster/src/main/paradox/index.md

```diff
@@ -0,0 +1,600 @@
+3. [Containerizing your application](Containerizing-your-application)
+3.1 [Containerizing your application - SBT](Containerizing-your-application---SBT)
+3.1.1 [SBT multi-module project](SBT-multi-module-project)
+3.1.2 [SBT single-module project](SBT-single-module-project)
+3.1.3 [Optional: Using a smaller Docker base image](Optional:-Using-a-smaller-Docker-base-image)
+3.2 [Containerizing your application - Maven](Containerizing-your-application---Maven)
+3.2.1 [Maven multi-module project](Maven-multi-module-project)
+3.2.2 [Maven single-module project](Maven-single-module-project)
+4. [Creating Kubernetes Service for Akka remoting](Creating-Kubernetes-Service-for-Akka-remoting)
+5. [Creating Kubernetes StatefulSet resource](Creating-Kubernetes-StatefulSet-resource)
+
+
+
+## 1. Naming your application
+
+In this guide we will refer to the application name as `myapp` - please feel free to substitute the application name with the actual name of the application you are going to deploy.
+
+We'd like to bring your attention to the application name as they would be used to define the service name in the Kubernetes StatefulSet and Kubernetes Service.
+
+The application name will also be referenced in the SBT or Maven build file, and as such the application name will form the Docker image name when containerized.
+
+
+## 2. Handling environment variables
+
+To establish Akka cluster within the Kubernetes container, we will be adding the following system properties when containerizing your application:
+
+```
+-Dakka.actor.provider=cluster
+-Dakka.remote.netty.tcp.hostname="$AKKA_REMOTING_BIND_HOST"
+-Dakka.remote.netty.tcp.port="$AKKA_REMOTING_BIND_PORT" -Dakka.cluster.seed-nodes.0="akka.tcp://${AKKA_ACTOR_SYSTEM_NAME}@${AKKA_SEED_NODE_HOST}:${AKKA_SEED_NODE_PORT}"
+-DactorSystemName=${AKKA_ACTOR_SYSTEM_NAME}
+```
+
+Some of the system properties values are derived from environment variables, i.e. `AKKA_SEED_NODE_HOST`. These environment variables will be supplied by the Kubernetes StatefulSet.
+
+| Environment Variable    | Description                         | Example Value |
+|-------------------------|-------------------------------------|---------------|
+| AKKA_REMOTING_BIND_HOST | The hostname of the Kubernetes Pod. | `myapp-2`     |
+| AKKA_REMOTING_BIND_PORT | The Akka remoting port.             | 2551          |
+| AKKA_SEED_NODE_HOST     | The hostname of the first container in the StatefulSet. | `myapp-0` |
+| AKKA_SEED_NODE_PORT     | The Akka remoting port of the seed node. In most cases this value should match `AKKA_REMOTING_BIND_PORT`. | 2551 |
+| AKKA_ACTOR_SYSTEM_NAME  | The name of the `ActorSystem` of your application. For the purpose of this guide, we'll match the `ActorSystem` name with the application name. | 'myapp' |
+
+
+The following code can be used by your application to set up the `ActorSystem` based on the name specified by `actorSystemName` system property:
```

Comment:
"can be used by your application" or "should be used"? If there are multiple ways to do it, state that and then say, for example, the following code ....